src/pipelinedefinition.cpp

```diff
@@ -411,13 +423,29 @@ class NodeValidator {
                 shape.size());
             return StatusCode::PIPELINE_NOT_ENOUGH_SHAPE_DIMENSIONS_TO_DEMULTIPLY;
         }
-        if (shape[1] != demultiplicatorNodeInfo.demultiplyCount.value()) {
-            SPDLOG_LOGGER_ERROR(modelmanager_logger, "Validation of pipeline: {} definition failed. Demultiply count: {} of node: {} does not match tensor second dimenson value: {}",
+        if (demultiplicatorNodeInfo.demultiplyCount.value() != 0) {
```

Comment:
This method might not be guaranteed that demultipllcatorNodeInfo is in fact demultiplicator. There might be exception thrown when calling demultplyCount.value() when containing std::nullopt.

Reply:
It is checked before calling this method. I added return OK just in case.

Reply:
I don't think this is checked before calling this method. We take node name from gatherFromNode and find node info with such name. Pipeline can be misconfigured in a way that this node is not actual demultiplexer node.